broker/tests/broker_server_test.cpp

```diff
@@ -0,0 +1,421 @@
+        ieee1905_1::CmduMessageRx cmdu_rx(rx_thread_buffer + sizeof(message::sUdsHeader),
+                                          sizeof(rx_thread_buffer));
+
+        ASSERT_TRUE(cmdu_rx.parse());
+        ASSERT_EQ(cmdu_rx.getMessageType(), ieee1905_1::eMessageType::TOPOLOGY_QUERY_MESSAGE);
+
+        // Send a TOPOLOGY_RESPONSE_MESSAGE message
+        LOG(DEBUG) << "Sending TOPOLOGY_RESPONSE_MESSAGE...";
+        ASSERT_TRUE(cmdu.create(0, ieee1905_1::eMessageType::TOPOLOGY_RESPONSE_MESSAGE) != nullptr);
+        ASSERT_TRUE(message_com::send_cmdu(&sock1, cmdu));
+        UTILS_SLEEP_MSEC(100);
+    });
+
+    static uint8_t tx_thread_buffer[message::MESSAGE_BUFFER_LENGTH];
+    std::thread tx_thread([&] {
+        // Wait for the RX thread to complete subscription
+        UTILS_SLEEP_MSEC(100);
+
+        // Create an empty CMDU
+        ((message::sUdsHeader *)(tx_thread_buffer))->magic = message::MESSAGE_MAGIC;
+        ieee1905_1::CmduMessageTx cmdu(tx_thread_buffer + sizeof(message::sUdsHeader),
+                                       sizeof(tx_thread_buffer));
+
+        // Subscribe to TOPOLOGY_RESPONSE_MESSAGE message
+        auto subscribe =
+            message_com::create_vs_message<beerocks_message::cACTION_BROKER_SUBSCRIBE>(cmdu);
+
+        ASSERT_TRUE(subscribe != nullptr);
+        ASSERT_TRUE(subscribe->alloc_msg_type_list());
+        *subscribe->msg_type_list(0) =
+            (uint32_t)ieee1905_1::eMessageType::TOPOLOGY_RESPONSE_MESSAGE;
+
+        // Connect to the broker and send the subscription message
+        SocketClient sock1(broker_uds_file);
+        ASSERT_TRUE(message_com::send_cmdu(&sock1, cmdu));
+
+        // Send a TOPOLOGY_QUERY_MESSAGE message
+        LOG(DEBUG) << "Sending TOPOLOGY_QUERY_MESSAGE...";
+        ASSERT_TRUE(cmdu.create(0, ieee1905_1::eMessageType::TOPOLOGY_QUERY_MESSAGE) != nullptr);
+        ASSERT_TRUE(message_com::send_cmdu(&sock1, cmdu));
+        UTILS_SLEEP_MSEC(100);
+
+        // Wait for TOPOLOGY_RESPONSE_MESSAGE message
+        SocketSelect select;
+        timeval timeout = {.tv_sec = 1, .tv_usec = 0}; // 1sec timeout
+        select.setTimeout(&timeout);
+        select.addSocket(&sock1);
+        if (select.selectSocket() <= 0) {
+            FAIL() << "Didn't receive a message from the broker in the specified timeout";
+        }
+
+        // Receive and validate the message
+        ASSERT_GT(sock1.readBytes(tx_thread_buffer, sizeof(tx_thread_buffer), true), 0);
+        ieee1905_1::CmduMessageRx cmdu_rx(tx_thread_buffer + sizeof(message::sUdsHeader),
+                                          sizeof(tx_thread_buffer));
+
+        ASSERT_TRUE(cmdu_rx.parse());
+        ASSERT_EQ(cmdu_rx.getMessageType(), ieee1905_1::eMessageType::TOPOLOGY_RESPONSE_MESSAGE);
+    });
+
+    rx_thread.join();
+    tx_thread.join();
+}
+
+} // namespace tests
+} // namespace broker
+} // namespace beerocks
+
+int main(int argc, char *argv[])
```

Comment:
Pleas move main() function to a separate file.

Reply:
Why?

Reply:
Because you'll probably end up adding more files with tests (one file per test fixture, in practice and most of the times, one file per class being tested).

Reply:
Replaced `main` with a `setup` test